src/continuum_absorption/absorption_ff_metal.jl

```diff
@@ -0,0 +1,458 @@
+end
+
+_Mg_I_ff(ν::Real, T::Real, ndens_MgII::Real, nₑ::Real) =
+    metal_ff_absorption_departure(ν, T, 1, ndens_MgII, nₑ, _departure_term_Mg_I_ff)
+
+
+"""
+    metal_ff_absorption(ν, T, Z, ni, ne)
+
+Computes the free-free linear absorption coefficient (in cm⁻¹) of a metal species for which a 
+departure term is specified.
+
+A free-free interaction is named as though the species interacting with the free electron had one 
+more bound electron (in other words it's named as though the free-electron and ion were bound 
+together). The `Z` argument and `ni` values should be specified for the species that actually
+participates in the reaction. For completeness 2 examples provided below:
+- Si I ff absorption: `ni` holds the number density of Si II, and `Z=1` (net charge of Si II)
+- Si II ff absorption: `ni` holds the number density of Si III, and `Z=2` (net charge of Si III)
+
+This function simply wraps the hydrogenic free-free calculation.
+
+# Arguments
+- `ν`: frequency in Hz
+- `T`: temperature in K
+- `Z::Integer`: the net charge of the ion species (that participates in the interaction).
+- `ni`: the number density of the ion species (that participates in the interaction) in cm⁻³.
+- `ne`: the number density of free electrons.
+"""
+metal_ff_absorption(ν, T, Z, ni, ne) = hydrogenic_ff_absorption(ν, T, Z, ni, ne)
+
+
+
+"""
+    all_metal_ff_absorption(ν::Real, T::Real, number_densities::Dict, ne::Real,
+                            exclude_species_with_departure_terms::Bool = false)
+
+Computes the free-free linear absorption coefficient (in cm⁻¹) of metal species. This only includes
+free-free absorption for species with a net positive charge of 1, 2, or 3. It also skips Hydrogenic
+atoms.
+
+# Arguments
+- `ν`: frequency in Hz
+- `T`: temperature in K
+- `number_densities` is a `Dict` mapping each `Species` to its number density
+- `ne`: the number density of free electrons.
+
+This uses the hydrogenic approximation for all species except for those that have departure terms.
+
+!!! note
+    This mostly exists for demonstrative purposes.
+
+    It might be useful to adopt this function for long-term use for efficiency reasons (even though
+    it departs from the format of all other continuum absorption calculations).  
+
+"""
+function all_metal_ff_absorption(ν::Real, T::Real, number_densities::Dict, ne::Real,
+                                 exclude_species_with_departure_terms::Bool = false)
+
+    error("I think we may need the user to pass in the partition functions for every species")
+
+    # recall, name of the free-free interaction differs from the name of the participating species
+    departure_dict = Dict([(species"He_II", _He_I_ff),
+                           (species"C_II",  _C_I_ff),
+                           (species"C_III", _C_II_ff),
+                           (species"Si_II", _Si_I_ff),
+                           (species"Mg_II", _Mg_I_ff)])
```

Comment:
One possible way to refactor this is to define this dictionary in the file where the `D` values are tabulated.  Alternatively, you could do a function `departure_from_hydrogenic(spec::Species)` with `if else`s to switch between the species.  You could then hide the interpolator objects behind a module and export that function only.  In this vision, this would replace `metal_ff_absorption_departure` and all the info in that docstring would be in the docstring for this function as well.